routers/web/repo/view.go

```diff
@@ -728,6 +728,54 @@ func checkHomeCodeViewable(ctx *context.Context) {
 	ctx.NotFound("Home", fmt.Errorf(ctx.Tr("units.error.no_unit_allowed_repo")))
 }
 
+func checkCitationFile(ctx *context.Context) {
+	// Check if repo is not empty and if we are on repo src
+	if ctx.Repo.Repository.IsEmpty {
+		return
+	}
+	entry, err := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)
+	if err != nil {
+		ctx.NotFoundOrServerError("Repo.Commit.GetTreeEntryByPath", git.IsErrNotExist, err)
+		return
+	}
+	if len(entry.Name()) != 0 {
```

Comment:
Why is this check here anyway? Looking at [`renderCode`](https://github.com/go-gitea/gitea/blob/9eb6c4a0dfaed86713952da19a8aa5079e97addb/routers/web/repo/view.go#L901) it assume that you can safely execute `tree, err := ctx.Repo.Commit.SubTree(ctx.Repo.TreePath)` without any necessary checks.